integration-tests/getAllCategoriesIntegration.test.js

```diff
@@ -0,0 +1,228 @@
+import request from "supertest";
+import mongoose from "mongoose";
+import { MongoMemoryServer } from "mongodb-memory-server";
+import JWT from "jsonwebtoken";
+import app from "../server.js";
+import { ObjectId } from "mongodb";
+import categoryModel from "../models/categoryModel.js";
+
+describe("Get All Categories Integration Tests", () => {
+  let mongoMemServer;
+  let category1, category2, category3, category4, category5;
+  const getAllCategoryPath = "/api/v1/category/get-category";
+  let token;
+
+  beforeAll(async () => {
+    mongoMemServer = await MongoMemoryServer.create();
+    await mongoose.connect(mongoMemServer.getUri());
+
+    category1 = await categoryModel.create({
+      name: "Test Category 1",
+      slug: "test-category-1",
+    });
+    category2 = await categoryModel.create({
+      name: "Test Category 2",
+      slug: "test-category-2",
+    });
+
+    category3 = await categoryModel.create({
+      name: "Test Category 3",
+      slug: "test-category-3",
+    });
+
+    category4 = await categoryModel.create({
+      name: "Test Category 4",
+      slug: "test-category-4",
+    });
+
+    category5 = await categoryModel.create({
+      name: "Test Category 5",
+      slug: "test-category-5",
+    });
+
+    // Create a mock JWT token for authentication tests
```

Comment:
Is there a need to add JWT since this controller does not pass through any middleware which uses JWT?

Reply:
I decided to demonstrate pairwise testing between authentication and getAllCategories to test the interplay between the two mechanisms 😃